validator/resource_value_pattern_test.rego

```diff
@@ -0,0 +1,187 @@
+package templates.gcp.GCPResourceValuePatternConstraintV1
+
+import data.test.fixtures.resource_value_pattern.assets as fixture_assets
+import data.test.fixtures.resource_value_pattern.constraints.list as fixture_constraints
+
+# Helper to lookup a constraint based on its name via metadata, not package
+lookup_constraint[name] = [c] {
+	c := fixture_constraints[_]
+	c.metadata.name = name
+}
+
+# Helper to execute constraints against assets
+find_violations[violation] {
+	asset := data.assets[_]
+	constraint := data.test_constraints[_]
+	issues := deny with input.asset as asset with input.constraint as constraint
+	total_issues := count(issues)
+	violation := issues[_]
+}
+
+# Helper to create a set of resource names from violations
+resource_names[name] {
+	# Not sure why I need this, data.violations was a array_set but unless
+	# casted as an array all evals of X[_] would fail.  Tested iterating sets in
+	# playground and they work fine, so I am not sure the problem here.
+	a := cast_array(data.violations)
+	i := a[_]
+	trace(sprintf("Violations %v", [i]))
+	name := i.details.resource
+}
+
+test_resource_value_pattern_optional_on_multiple_resources {
+	# Arrange
```

Comment:
nit: are the "Arrange", "Act" and "Assert" comments useful? They don't seem to convey any additional information?